File: issta2018artifacts/minimized-bugs/fixed-key/Session.java
// simplified version of bug fixed in Daikon commit cc9edee13

public class Session {
    /*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
    public static void do_session(InputStream is, byte[] buf) {
	int pos = is.read(buf);
        if (pos != -1) {
            //:: error: argument.type.incompatible
            String actual = new String(buf, 0, pos);
        }
    }

    private static class InputStream {
        /*@ public normal_behavior
@ requires true;
@ ensures \result >= -1;
@*/
        public int read(byte[] buf) {
            return -1;
        }
    }
}

File: issta2018artifacts/minimized-bugs/buggy-clousot/Daikon/Session.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.Contracts;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

// simplified version of bug fixed in Daikon commit cc9edee13

abstract class InputStream
{
    public virtual int read(byte[] buf)
    {
        Contract.Ensures(Contract.Result<int>() >= -1 && Contract.Result<int>() <= buf.Length);
        return -1;
    }
}

class Session
{
    public static void do_session(InputStream @is, byte[] buf)
    {
        Contract.Requires(buf != null);
        int pos = @is.read(buf);
        //:: error: argument.type.incompatible
        string actual = Encoding.ASCII.GetString(buf, 0, pos);
    }
}

